File: LeetCode/Easy/2022-convert-1d-array-into-2d-array/2022-convert-1d-array-into-2d-array.java
class Solution {
    public int[][] construct2DArray(int[] original, int m, int n) {
        if (original.length != m * n) return new int[0][0];
        
        int[][] ret = new int[m][n];
        
        for (int row = 0; row < m; ++row)
            for (int col = 0; col < n; ++col)
                ret[row][col] = original[row * n + col];
        
        return ret;
    }
}

File: LeetCode/Easy/2022-convert-1d-array-into-2d-array/2022-convert-1d-array-into-2d-array.go
func construct2DArray(original []int, m int, n int) [][]int {
    if len(original) != m * n { return [][]int{} }

    ret := make([][]int, 0)

    for row := 0; row < m; row++ {
        ret = append(ret, original[row * n : (row + 1) * n]) 
    }

    return ret
}